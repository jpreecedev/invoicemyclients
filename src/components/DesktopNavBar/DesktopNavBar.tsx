import React from 'react'
import { Responsive, Segment } from 'semantic-ui-react'

import hero from '../../assets/images/landing-hero.jpg'
import { MenuBar } from '../MenuBar'

interface DesktopNavBarProps {
  heading: React.ReactElement
}

const getWidth = () => {
  return typeof window === 'undefined'
    ? (Responsive.onlyTablet.minWidth as number)
    : window.innerWidth
}

const DesktopNavBar: React.FC<DesktopNavBarProps> = ({ children, heading }) => {
  return (
    <Responsive getWidth={() => getWidth()} minWidth={Responsive.onlyTablet.minWidth}>
      <Segment
        inverted
        textAlign="center"
        style={{ minHeight: 700, padding: '1em 0em', backgroundImage: `url(${hero})` }}
        vertical
      >
        <MenuBar inverted={true} />
        {heading}
      </Segment>

      {children}
    </Responsive>
  )
}

export { DesktopNavBar }
