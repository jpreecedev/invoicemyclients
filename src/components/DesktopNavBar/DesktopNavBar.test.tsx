import React from 'react'
import { render } from '@testing-library/react'
import { DesktopNavBar } from './DesktopNavBar'

describe('<DesktopNavBar /> tests', () => {
  it('should match snapshot', () => {
    const heading = <h1>Heading</h1>

    expect(
      render(
        <DesktopNavBar heading={heading}>
          <div>Foo</div>
        </DesktopNavBar>
      ).container
    ).toMatchSnapshot()
  })
})
