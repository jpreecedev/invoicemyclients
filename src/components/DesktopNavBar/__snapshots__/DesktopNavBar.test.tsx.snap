// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DesktopNavBar /> tests should match snapshot 1`] = `
<div>
  <div>
    <div>
      <div
        class="ui inverted vertical center aligned segment"
        style="min-height: 700px; padding: 1em 0em; background-image: url(landing-hero.jpg);"
      >
        <div
          class="ui large inverted pointing secondary menu"
        >
          <div
            class="ui container"
          >
            <a
              class="item"
            >
              Features
            </a>
            <a
              class="item"
            >
              Pricing
            </a>
            <div
              class="right item"
            >
              <a
                class="ui primary button"
                role="button"
                style="margin-left: 0.5em;"
              >
                Sign Up
              </a>
            </div>
          </div>
        </div>
        <h1>
          Heading
        </h1>
      </div>
    </div>
    <div>
      Foo
    </div>
  </div>
</div>
`;
